File: Python/BOJ/1931 회의실 배정.py
import sys
N = int(sys.stdin.readline())
# 1번 인덱스를 기준으로 정렬
meeting = sorted([list(map(int,sys.stdin.readline().split())) for _ in range(N)], key=lambda x:(x[1],x[0]))
cnt = 1 # 가능 횟수 체크
temp = meeting[0][1] # 맨 앞시작 값
for i in range(1,N):
    if meeting[i][0] >= temp: # 시작시간이 현재 끝나는 시간보다 크면
        temp = meeting[i][1] # 끝나는 시간 저장하고
        cnt += 1 #횟수 증가
print(cnt)

File: Java/BOJ/1931 회의실 배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[][] meeting = new int[N][2];
        
        for(int i = 0; i < N; i++) {
        	StringTokenizer st = new StringTokenizer(br.readLine()," ");
        	meeting[i][0] = Integer.parseInt(st.nextToken());
        	meeting[i][1] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(meeting, (a,b) -> a[1] == b[1] ? a[0] - b[0] : a[1] - b[1]);
        
        int cnt = 1;
        int temp = meeting[0][1];
        for(int i = 1; i< N; i++) {
        	if(meeting[i][0] >= temp) {
        		temp = meeting[i][1];
        		cnt++;
        	}
        }
        System.out.println(cnt);
    }
}